secboot/secboot_tpm.go

```diff
@@ -238,14 +238,40 @@ func newAuthRequestor() (sb.AuthRequestor, error) {
 	)
 }
 
-// TODO: consider moving this to secboot
-func readKeyFileImpl(keyfile string) (*sb.KeyData, *sb_tpm2.SealedKeyObject, error) {
+type keyLoader interface {
+	// LoadedKeyData keeps track of keys in KeyData format.
+	LoadedKeyData(kd *sb.KeyData)
+	// LoadedSealedKeyObject keeps track of sealed key object for
+	// legacy TPM This is useful for resealing. For unlocking, a
+	// matching KeyData will also be emitted.
+	LoadedSealedKeyObject(sko *sb_tpm2.SealedKeyObject)
+	// LoadedFDEHookKeyV1 keeps track of sealed key object for
+	// legacy FDE hooks. In this case no KeyData is emitted.
+	LoadedFDEHookKeyV1(sk []byte)
+}
+
+type defaultKeyLoader struct {
```

Comment:
then this is also defaultLoadedKey?

Reply:
I have put a todo to the `keyLoader` interface to change it.

I am not sure that `defaultLoadedKey` makes total sense here since we create it empty. So it is not loaded yet. But it will make sense once we get rid of the interface and we return it.